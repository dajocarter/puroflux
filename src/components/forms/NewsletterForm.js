import React, { Component } from 'react'
import { FormGroup, FormControl, FormLabel, Button } from 'react-bootstrap'

import 'bootstrap/dist/css/bootstrap.css'

export default class NewsletterForm extends Component {
  render() {
    return (
      <form
        name="newsletter"
        method="POST"
        action="https://formspree.io/dajocarter@gmail.com"
      >
        <p>ENTER YOUR NAME AND EMAIL TO GET NEWS & MORE!</p>
        <FormGroup controlId="name">
          <FormLabel>Name</FormLabel>
          <FormControl type="text" name="name" placeholder="Your name" />
        </FormGroup>
        <FormGroup controlId="email">
          <FormLabel>Email</FormLabel>
          <FormControl type="email" name="_replyto" placeholder="Your email" />
        </FormGroup>
        <input
          type="hidden"
          name="_subject"
          value="New Newsletter Signup Form Submission"
        />
        <input type="text" name="_gotcha" style={{ display: 'none' }} />
        <Button type="submit" variant="light">
          Submit
        </Button>
      </form>
    )
  }
}
