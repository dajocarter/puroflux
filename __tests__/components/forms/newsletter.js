import React from 'react'
import renderer from 'react-test-renderer'

import NewsletterForm from '../../../src/components/forms/newsletter'

describe('Newsletter Form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<NewsletterForm />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
