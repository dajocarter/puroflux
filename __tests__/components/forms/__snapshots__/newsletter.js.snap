// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter Form renders correctly 1`] = `
<form
  data-netlify="true"
  data-netlify-honeypot="bot-field"
  method="POST"
  name="Newsletter"
>
  <input
    name="form-name"
    type="hidden"
    value="Newsletter"
  />
  <p
    className="d-none"
  >
    <label>
      Donâ€™t fill this out:
       
      <input
        name="bot-field"
        type="text"
      />
    </label>
  </p>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="form-control"
      id="name"
      name="name"
      placeholder="Your name"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="form-control"
      id="email"
      name="_replyto"
      placeholder="Your email"
      type="email"
    />
  </div>
  <input
    name="_subject"
    type="hidden"
    value="New submission from Newsletter form."
  />
  <button
    className="btn btn-light"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
